### PR TITLE
Empty state for no search results

### PR DESCRIPTION
**Pull request by cdolm92**
## WHAT

Added empty state for no search results. Users are now informed when a search result is not found. 

search-empty-state

### CODE REVIEW

**Comment by cdolm92** [comment]
Just waiting to update strings now

File: Library/ViewModels/SearchViewModel.swift [hunk 1]
```diff
@@ -123,6 +126,12 @@ public final class SearchViewModel: SearchViewModelType, SearchViewModelInputs,
       .map { showPopular, popular, searchResults in showPopular ? popular : searchResults }
       .skipRepeats(==)
 
+    let shouldShowNoSearchResults = Signal.combineLatest(isLoading, .merge(clears, projects))
```
Comment (mbrandonw) [inline Library/ViewModels/SearchViewModel.swift]: im a liiiiil bit worried about this combine latest and over-emission problem. wondering if the `clears` are necessary?

maybe we could just add a few more tests to subside my fears. perhaps the situation of typing in a query that brings back projects, and then a new query that doesnt have any results, and making sure this emits properly

File: Library/ViewModels/NoSearchResultsCellViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import KsApi
+import ReactiveSwift
+import Result
+
+public protocol NoSearchResultsCellViewModelInputs {
+  func configureWith(param: DiscoveryParams)
+}
+
+public protocol NoSearchResultsCellViewModelOutputs {
+  var searchTermNotFound: Signal<String, NoError> { get }
```
Comment (mbrandonw) [inline Library/ViewModels/NoSearchResultsCellViewModel.swift]: we could be more explicit by calling this `searchTermNotFoundLabelText`

File: Library/ViewModels/SearchViewModel.swift [hunk 3]
```diff
@@ -123,7 +123,8 @@ public final class SearchViewModel: SearchViewModelType, SearchViewModelInputs,
       requestFromParams: requestFromParamsWithDebounce,
       requestFromCursor: { AppEnvironment.current.apiService.fetchDiscovery(paginationUrl: $0) })
 
-    self.projects = Signal.combineLatest(self.isPopularTitleVisible, popular, .merge(clears, paginatedProjects))
+    self.projects =
+      Signal.combineLatest(self.isPopularTitleVisible, popular, .merge(clears, paginatedProjects))
       .map { showPopular, popular, searchResults in showPopular ? popular : searchResults }
       .skipRepeats(==)
```
Comment (mbrandonw) [inline Library/ViewModels/SearchViewModel.swift]: small comment, i think this indentation would read a bit better:

```swift
    self.projects = Signal.combineLatest(
      self.isPopularTitleVisible, popular, 
      .merge(clears, paginatedProjects)
      )
      .map { showPopular, popular, searchResults in showPopular ? popular : searchResults }
      .skipRepeats(==)
```

**Pull request merged**